feat: Adds separate Android TV and iOS TV build workflows

Gauvino opened the pull request:
Splits build workflows to support both phone and TV variants for Android and iOS platforms.

Creates dedicated jobs for TV builds that use specific prebuild commands and environment variables. Updates workflow names to reflect dual-platform support and improves artifact naming to distinguish between phone and TV builds.

Removes JDK setup from Android workflow as it's handled by build scripts. Updates cache action versions and standardizes step descriptions across workflows.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Chores**
  * Updated Android and iOS build workflows to separately build and upload artifacts for both Phone and TV platforms.
  * Renamed and split build jobs for clearer distinction between Phone and TV targets.
  * Improved artifact naming and workflow step clarity for easier identification of build outputs.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Gauvino commented on .github/workflows/build-android.yml:
```diff
@@ -32,48 +32,97 @@ jobs:
+        uses: actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02 # v4.6.2
+        with:
+          name: streamyfin-android-tv-apk-${{ env.DATE_TAG }}
+          path: android/app/build/outputs/apk/release/*.apk
```
It's not the first time i get this suggestion but we don't need .aab @fredrikburmester

Gauvino commented:
@storm1er could you review this and tell me ?

lancechant commented on .github/workflows/build-android.yml:
```diff
@@ -32,48 +32,97 @@ jobs:
+            ${{ runner.os }}-android-deps-
+
+      - name: 🛠️ Generate project files (TV)
+        run: bun run prebuild:tv
```
@Gauvino I don't see the EXPO_TV env here for android? but it's there on IOS?

Gauvino replied:
@lancechant take a look at this line inside package.json:
```
"prebuild": "cross-env EXPO_TV=0 bun run clean",
"prebuild:tv": "cross-env EXPO_TV=1 bun run clean",
```

lancechant replied:
Would maybe be nice to have the commands be like build prod in the package json for the future to keep uniformity

Gauvino replied:
what do you mean by this ?

lancechant replied:
For the ios path, bun run build-prod as a script in the package.json (just as an example) if that makes sense? then you only have to change that file in the future

Gauvino replied:
maybe just do a r and i'll review if you can to see what you mean

lancechant replied:
Will do at a later point. But happy for it to go through now, can't approve this sadly 🤣

Gauvino committed:
feat: Adds separate Android TV and iOS TV build workflows

Splits build workflows to support both phone and TV variants for Android and iOS platforms.

Creates dedicated jobs for TV builds that use specific prebuild commands and environment variables. Updates workflow names to reflect dual-platform support and improves artifact naming to distinguish between phone and TV builds.

Removes JDK setup from Android workflow as it's handled by build scripts. Updates cache action versions and standardizes step descriptions across workflows.

MajorP93 commented:
Android TV release soon? 👀

Gauvino commented:
> Android TV release soon? 👀

Maybe 🫣
